README.md

```diff
@@ -265,6 +294,80 @@ final StepProperty multiChooser = new StepProperty()
     .withSelectAllOption(false);
 ```
 
+##### Auto select column based on data tag
+Data Studio allows tagging columns with additional label. Refer to [Data Tagging](https://www.edq.com/documentation/aperture-data-studio/user-guide/#data-tagging) for more info. 
+
+You can make the column chooser step property to automatically select a column tagged with specific label (e.g. 'Phone') once the custom step connected with an input.
+
+![Tagged Column](images/tagged.png)
+
+```java
+public class ColumnTagStep extends StepConfiguration {
+
+    private static final String TAG_PHONE = "Phone";
+    
+    public ColumnTagStep() {
+        setStepDefinitionName("Custom - Column Tag");
+        setStepDefinitionIcon("INFO");
+
+        final StepProperty spPhone = new StepProperty()
+                .ofType(StepPropertyType.COLUMN_CHOOSER)
+                .withStatusIndicator(sp -> () -> isPhoneArgOk(sp))
+                .withIconTypeSupplier(sp -> () -> isPhoneArgOk(sp) ? "OK" : "ERROR")
+                .withArgTextSupplier(sp -> () -> getPhoneArgText(sp))
+                .havingInputNode(() -> "input0")
+                .havingOutputNode(() -> "output0")
+                .validateAndReturn();
+        setStepProperties(Arrays.asList(spPhone));
+
+        setStepOutput(new ColumnTagOutput());
+    }
+
+    private boolean isPhoneArgOk(final StepProperty sp) {
+        autoSelectTaggedColumn(sp, TAG_PHONE);
+        return sp.allowedValuesProvider != null && sp.getValue() != null;
+    }
+
+    private String getPhoneArgText(final StepProperty sp) {
+        autoSelectTaggedColumn(sp, TAG_PHONE);
+        if (sp.allowedValuesProvider == null) {
+            return "<Connect an input>";
+        } else {
+            return sp.getValue() == null ? "<Select a phone column>" : sp.getValue().toString();
+        }
+    }
+```
+
+Code above is very similar to normal step configuration, except it will call `autoSelectTaggedColumn` method every time it updates UI with `withStatusIndicator`, `withIconTypeSupplier` and `withArgTextSupplier` methods.
```

Comment:
```suggestion
The code above is very similar to normal step configuration, except it will call the `autoSelectTaggedColumn` method every time it updates the UI with `withStatusIndicator`, `withIconTypeSupplier` and `withArgTextSupplier` methods.
```